homeassistant/helpers/httpclient.py

```diff
@@ -0,0 +1,110 @@
+"""Helper for aiohttp webclient stuff."""
+import asyncio
+
+import aiohttp
+
+from homeassistant.core import callback
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP
+
+
+DATA_CONNECTOR = 'aiohttp_connector'
+DATA_CONNECTOR_NOTVERIFY = 'aiohttp_connector_notverify'
+DATA_CLIENTSESSION = 'aiohttp_clientsession'
+DATA_CLIENTSESSION_NOTVERIFY = 'aiohttp_clientsession_notverify'
+
+
+def get_clientsession(hass, verify_ssl=True):
+    """Return default aiohttp ClientSession.
+
+    This method must be run in the event loop.
+    """
+    if verify_ssl:
+        key = DATA_CLIENTSESSION
+    else:
+        key = DATA_CLIENTSESSION_NOTVERIFY
+
+    if key not in hass.data:
+        connector = _async_get_connector(hass, verify_ssl)
+        clientsession = aiohttp.ClientSession(
+            loop=hass.loop,
+            connector=connector
+        )
+        _async_register_clientsession_shutdown(hass, clientsession)
+        hass.data[key] = clientsession
+
+    return hass.data[key]
+
+
+def create_clientsession(hass, verify_ssl=True, auto_cleanup=True, **kwargs):
+    """Create a new ClientSession with kwargs, i.e. for cookies.
+
+    Is auto_cleanup False, you need call detach() after this session will be
+    unused. True it do it self on homeassistant_stop.
+
+    This method must be run in the event loop.
+    """
+    connector = _async_get_connector(hass, verify_ssl)
+
+    clientsession = aiohttp.ClientSession(
+        loop=hass.loop,
+        connector=connector,
+        **kwargs
+    )
+
+    if auto_cleanup:
+        _async_register_clientsession_shutdown(hass, clientsession)
+
+    return clientsession
+
+
+def _async_register_clientsession_shutdown(hass, clientsession):
+    """Register ClientSession close on homeassistant shutdown.
+
+    This method must be run in the event loop.
+    """
+    @callback
+    def _async_close_websession():
+        """Close websession."""
+        clientsession.detach()
+
+    hass.bus.async_listen_once(
+        EVENT_HOMEASSISTANT_STOP, _async_close_websession)
+
+
+def _async_get_connector(hass, verify_ssl=True):
```

Comment:
The main thing of aiohttp is not the ClientSession it is the connector. We should only have 2 connector (with ssl, without) that is also the connection pool and all the mechanic of that. I need this on `create_clientsession` and `get_clientsession`